enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,562 @@
+this feature part of the [rebootless updates](https://github.com/openshift/machine-config-operator/blob/94fc0354f154e2daa923dd96defdf86448b2b327/docs/MachineConfigDaemon.md?plain=1#L147)
+list in MCO. That way if you wanted to, for example, scale out a new machine
+you would just add its config to the existing machine-config, roll that out,
+then deploy the new node. These configs will only be applied at deploy time
+anyway since Kubernetes-NMState will be used for day 2 changes, so it shouldn't
+ever be necessary to reboot a node for changes to these configs.
+
+I believe we would need to add logic for these new config to
+[this code](https://github.com/openshift/machine-config-operator/blob/a41f5af837d95e0fc4f59e4497447a26acaf5bc2/pkg/daemon/update.go#L328)
+in MCO to make changes without rebooting.
+
+Scaleout in this implementation (using static IPs on baremetal to demonstrate
+as many parts of the process as possible) would look something like this:
+
+* Add the new node's configuration to the machine-config for its role.
+  * `oc edit mc 10-br-ex-worker`
+* Create and apply a secret containing the minimal static IP configuration
+  needed for the node to pull ignition.
+  * `oc apply -f ./extraworker-secret.yaml`
+* Create and apply a BareMetalHost resource with the network secret attached
+  via the `preprovisioningNetworkDataName` field.
+```
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+spec:
+...
+  preprovisioningNetworkDataName: ostest-extraworker-0-network-config-secret
+```
+* Scale out the machineset to deploy the new node.
+  * `oc project openshift-machine-api`
+    `oc get machinesets`
+    `oc scale machineset [name of machineset] --replicas=[number of nodes desired after scaling]`
+
+The node will initially boot with the network configuration specified in
+`preprovisioningNetworkDataName` and then have the configuration in the
+machine-config applied.
+
+### API Extensions
+
+This will not directly modify the API of OpenShift. The API for this feature is
+implemented in Kubernetes-NMState, and this does not require any specific
+modifications in that project at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+Because this gives the deployer a great deal of control over the bridge
+configuration for OVNK, they will have the ability to configure it incorrectly,
+which may cause problems either on initial deployment or at any point after.
+We will want to ensure that must-gather is collecting enough information about
+the bridge configuration for us to determine whether it is compliant with our
+guidelines, and we must have clear instructions on what is required for br-ex.
```

Comment:
To reduce this risk we should document nmstate examples to create br-ex manually for differente scenarios (single-nic, multi-nic, bond + bridge + vlans, etc...)

We have to be super careful with MTUs though.

Reply:
Yep, probably similar to what we have for kubernetes-nmstate today. We'll include examples of some common configs.